src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java

```diff
@@ -408,6 +478,117 @@ public ElasticsearchSinkConnectorConfig(Map<String, String> props) {
     super(CONFIG, props);
   }
 
+  public boolean isAuthenticatedConnection() {
+    return username() != null && password() != null;
+  }
+
+  public int batchSize() {
+    return getInt(BATCH_SIZE_CONFIG);
+  }
+
+  public BehaviorOnMalformedDoc behaviorOnMalformedDoc() {
+    return BehaviorOnMalformedDoc.valueOf(
+        getString(BEHAVIOR_ON_MALFORMED_DOCS_CONFIG).toUpperCase()
+    );
+  }
+
+  public BehaviorOnNullValues behaviorOnNullValues() {
+    return BehaviorOnNullValues.valueOf(getString(BEHAVIOR_ON_NULL_VALUES_CONFIG).toUpperCase());
+  }
+
+  public int connectionTimeoutMs() {
+    return getInt(CONNECTION_TIMEOUT_MS_CONFIG);
+  }
+
+  public boolean createIndicesAtStart() {
+    return getBoolean(CREATE_INDICES_AT_START_CONFIG);
+  }
+
+  public boolean dropInvalidMessage() {
+    return getBoolean(DROP_INVALID_MESSAGE_CONFIG);
+  }
+
+  public long flushTimeoutMs() {
+    return getLong(FLUSH_TIMEOUT_MS_CONFIG);
+  }
+
+  public boolean ignoreKey() {
+    return getBoolean(IGNORE_KEY_CONFIG);
+  }
+
+  public Set<String> ignoreKeyTopics() {
+    return new HashSet<>(getList(IGNORE_KEY_TOPICS_CONFIG));
+  }
+
+  public boolean ignoreSchema() {
+    return getBoolean(IGNORE_SCHEMA_CONFIG);
+  }
+
+  public Set<String> ignoreSchemaTopics() {
+    return new HashSet<>(getList(IGNORE_SCHEMA_TOPICS_CONFIG));
+  }
+
+  public long lingerMs() {
+    return getLong(LINGER_MS_CONFIG);
+  }
+
+  public int maxBufferedRecords() {
+    return getInt(MAX_BUFFERED_RECORDS_CONFIG);
+  }
+
+  public int maxInFlightRequests() {
+    return getInt(MAX_IN_FLIGHT_REQUESTS_CONFIG);
+  }
+
+  public int maxRetries() {
+    return getInt(MAX_RETRIES_CONFIG);
+  }
+
+  public Password password() {
+    return getPassword(CONNECTION_PASSWORD_CONFIG);
+  }
+
+  public int readTimeoutMs() {
+    return getInt(READ_TIMEOUT_MS_CONFIG);
+  }
+
+  public long retryBackoffMs() {
+    return getLong(RETRY_BACKOFF_MS_CONFIG);
+  }
+
+  @Deprecated
+  public Map<String, String> topicToIndexMap() {
+    return parseMapConfig(getList(TOPIC_INDEX_MAP_CONFIG));
+  }
+
+  public String type() {
+    return getString(TYPE_NAME_CONFIG);
+  }
+
+  public Set<String> urls() {
```

Comment:
It would be great if we could use a more descriptive name for this getter, ex. `connectionUrls`. Let's also make sure `getList` doesn't eliminate duplicates already.

Reply:
it doesn't